### PR TITLE
FINERACT-637: Implementing Trial balance job

### PR DESCRIPTION
**Pull request by wkk91193**
[Fineract-637](https://issues.apache.org/jira/browse/FINERACT-637)
Creating a consolidated entry in the trial balance table for each day with the aggregation of all journal entries for that given day by grouping by GL account, branch, transaction date and updating the closing balance. As for which the results can be used for any reporting service or MIS requirements of MFIs or for merely reconciliation requirements

### CODE REVIEW

**Comment by avikganguly01** [comment]
Include JIRA ticket in PR name. If no ticket exists, create one.

File: fineract-provider/src/main/resources/sql/migrations/core_db/V343__m_trial_balance_table.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	`office_id` BIGINT(20) NOT NULL,
+	`account_id` BIGINT(20) NOT NULL,
+	`amount` DECIMAL(19,6) NOT NULL,
+	`entry_date` DATE NOT NULL,
+	`created_date` DATE NULL,
+	`closing_balance` DECIMAL(19,6) NOT NULL
+)
+COLLATE='latin1_swedish_ci'
+ENGINE=InnoDB;
+
```
Comment (avikganguly01) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V343__m_trial_balance_table.sql]: Remove collate and engine details.

File: fineract-provider/src/main/resources/sql/migrations/core_db/V343__m_trial_balance_table.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+INSERT INTO `job` (`name`, `display_name`, `cron_expression`, `create_time`, `task_priority`, `group_name`, `previous_run_start_time`, `next_run_time`, `job_key`, `initializing_errorlog`, `is_active`, `currently_running`, `updates_allowed`, `scheduler_group`, `is_misfired`) VALUES ('Update Trial Balance Details', 'Update Trial Balance Details', '0 1 0 1/1 * ? *', now(), 5, NULL, NULL, NULL, NULL, NULL, 1, 0, 1, 0, 0);
+
+ALTER TABLE acc_gl_journal_entry ADD transaction_date date;
+
```
Comment (avikganguly01) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V343__m_trial_balance_table.sql]: Also add index on transaction_date column.

**Review by avikganguly01 — APPROVED** [review]

**Comment by wkk91193** [comment]
@avikganguly01 
Removing collate,engine details and adding index to transaction_date done!

**Comment by ShruthiRajaram** [comment]
@wkk91193   kindly squash commits.

File: fineract-provider/src/main/resources/sql/migrations/core_db/V343__m_trial_balance_table.sql [hunk 3]
```diff
@@ -0,0 +1,35 @@
+--
```
Comment (ShruthiRajaram) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V343__m_trial_balance_table.sql]: V_343 file is added to the code already. So kindly use the next available migration script number

**Review by ShruthiRajaram — CHANGES_REQUESTED** [review]

**Review by ShruthiRajaram — APPROVED** [review]

**Pull request merged**